postgres/src/schema.rs

```diff
@@ -71,19 +89,45 @@ pub type TableId = u32;
 /// and the schemas of all its columns.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct TableSchema {
-    /// The fully qualified name of the table
-    pub table_name: TableName,
     /// The PostgreSQL OID of the table
-    pub table_id: TableId,
+    pub id: Oid,
+    /// The fully qualified name of the table
+    pub name: TableName,
     /// The schemas of all columns in the table
     pub column_schemas: Vec<ColumnSchema>,
 }
 
 impl TableSchema {
+    pub fn new(id: Oid, name: TableName, column_schemas: Vec<ColumnSchema>) -> Self {
+        Self {
+            id,
+            name,
+            column_schemas,
+        }
+    }
+
     /// Returns whether the table has any primary key columns.
     ///
     /// This method checks if any column in the table is marked as part of the primary key.
     pub fn has_primary_keys(&self) -> bool {
         self.column_schemas.iter().any(|cs| cs.primary)
     }
 }
+
+impl PartialOrd for TableSchema {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for TableSchema {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.name.cmp(&other.name)
+    }
+}
+
+impl Borrow<Oid> for TableSchema {
+    fn borrow(&self) -> &Oid {
+        &self.id
+    }
+}
```

Comment:
Implementing `Borrow` here doesn't make sense here because it's mainly for types to work well as keys for `HasMaps`.

Reply:
I did forget to remove it, thanks for noticing. I had it while generalizing the state store but I decided to not do it anymore.